### ISSUE TITLE
redaction is not working for wild cards after level 2

### ISSUE THREAD

**Issue by gurjotkaur20** [pinojs/pino#1708]
Issue: redaction is working upto level 2 when we use wildcards like ..* but after level 2, it is not working.

Here is the code where I want to redact field at level 3:
`
const logger = require('pino')({
    redact: {
        paths: [
            'entityPanNumber',
            '*.entityPanNumber',
            '*.*.entityPanNumber',
            '*.*.*.entityPanNumber'
        ],
        censor: '*****'
    }
  })

logger.info('temp: %o', { "entityPanNumber": "DEF" });
logger.info('temp: %o', { "1": { "entityPanNumber": "DEF" } });
logger.info('temp: %o', { "1": { "2": { "entityPanNumber": "DEF" } } });
logger.info('temp: %o', { "1": { "2": { "3": { "entityPanNumber": "DEF" } } }});
`

The output is:
`{"level":30,"time":1683219381177,"pid":166336,"hostname":"ThinkPadT480s","msg":"temp: {\"entityPanNumber\":\"*****\"}"}
{"level":30,"time":1683219381179,"pid":166336,"hostname":"ThinkPadT480s","msg":"temp: {\"1\":{\"entityPanNumber\":\"*****\"}}"}
{"level":30,"time":1683219381180,"pid":166336,"hostname":"ThinkPadT480s","msg":"temp: {\"1\":{\"2\":{\"entityPanNumber\":\"*****\"}}}"}
{"level":30,"time":1683219381182,"pid":166336,"hostname":"ThinkPadT480s","msg":"temp: {\"1\":{\"2\":{\"3\":{\"entityPanNumber\":\"DEF\"}}}}"}
`

**Comment by mcollina** [comment]
Thanks for reporting! I don't think we have the bandwidth to take on refactroing the redaction. A bugfix would be highly welcomed.

**Comment by gurjotkaur20** [comment]
> Thanks for reporting! I don't think we have the bandwidth to take on refactroing the redaction. A bugfix would be highly welcomed.

The issue is in fast-redact which is being used in pinojs. Let me check for the fix. 
Few follow-up queries:

- Should I raise the PR in fast-redact?
- If yes then we need to wait for fast-redact to pick up the PR and need to wait for the new version. Is there any way that I can do the change directly in pino js to avoid delays?

**Comment by mcollina** [comment]
Yes, it should be in fast-redact. I have the publish bit on fast-redact.

**Comment by gurjotkaur20** [comment]
> Yes, it should be in fast-redact. I have the publish bit on fast-redact.

I have raised PR in fast-redact. Please check.

**Comment by gurjotkaur20** [comment]
@mcollina I have raised PR in fast-redact. Could you please check?

**Comment by mcollina** [comment]
In case anybody is looking for the PR, it's available at https://github.com/davidmarkclements/fast-redact/pull/58
